ndcube/ndcube.py

```diff
@@ -201,10 +203,25 @@ def __init__(self, data, wcs, uncertainty=None, mask=None, meta=None,
                   extra_coords, self.missing_axis, data.shape)
         else:
             self._extra_coords_wcs_axis = None
-        # Initialize NDCube.
+        
+        # Enforce that the WCS object is a low_level_wcs object, complying APE14
+        if not isinstance(wcs, BaseLowLevelWCS):
+            raise TypeError(f'Expected a low_level_wcs object, got {type(wcs)}')
+        else:
+            # If the WCS object is low_level_wcs object, convert it into SlicedLowLevelWCS object for sanity
+            self.low_level_wcs = SlicedLowLevelWCS(wcs, [])
```

Comment:
I didn't test this as a slice in Astropy :thinking:

Reply:
That should return the same object, but wrapped as `SlicedLowLevelWCS` :sweat_smile:

Reply:
Yeah, and I would imagine it would work. I just mean when we implemented `SlicedLowLevelWCS` I don't think I wrote a unittest for `[]`